나민혁 1차 과제 제출

NaMinhyeok opened the pull request:
<!--
  템플릿은 아직 PR 작성이 익숙하지 않으신 분들을 위해서 제공하는 가이드입니다!
  리뷰어 또는 이 PR을 보게 될 다른 사람들이 이 PR을 보는데 참고할 수 있는 내용이 있다면 포함해서 작성해주시면 됩니다.
-->

## 📌 과제 설명 <!-- 어떤 걸 만들었는지 대략적으로 설명해주세요 -->

## 👩‍💻 요구 사항과 구현 내용 <!-- 기능을 Commit 별로 잘개 쪼개고, Commit 별로 설명해주세요 -->

## ✅ 피드백 반영사항  <!-- 지난 코드리뷰에서 고친 사항을 적어주세요. 재PR 시에만 사용해 주세요! (재PR 아닌 경우 삭제) -->

## ✅ PR 포인트 & 궁금한 점 <!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
객체 지향 설계를 하려고 노력했습니다. 패키지간 의존성 또한 생기지 않도록 노력하였습니다.
테스트코드를 모두 작성하려고 노력하였습니다.

NaMinhyeok committed:
chore: 테스트 환경 구성 및 docs, db 설정

NaMinhyeok committed:
Merge pull request #1 from NaMinhyeok/chore/프로젝트-기본-세팅추가

[chore] 테스트 환경 구성 및 docs, db 설정

NaMinhyeok committed:
feat: 엔티티 구성

NaMinhyeok committed:
feat: 공통 응답 및 보일러플레이트 코드 생성

NaMinhyeok committed:
Merge pull request #2 from NaMinhyeok/feat/엔티티-및-보일러플레이트코드-구현

[feat] 엔티티 및 보일러플레이트코드 구현

NaMinhyeok committed:
docs: 신규 상품 등록 API 문서화

NaMinhyeok committed:
docs: 상품 삭제 API 문서화

응답 데이터에 id를 넘겨주고 싶은데 잘 안됨 해결해야됨

NaMinhyeok committed:
docs: 상품 CRUD API 명세서 작성

NaMinhyeok committed:
docs: 주문 API 명세 작성

NaMinhyeok committed:
docs: 명세 제약사항 추가

NaMinhyeok committed:
Merge pull request #3 from NaMinhyeok/docs/api명세서-작성

[docs] API 명세서 작성

NaMinhyeok committed:
feat : 신규 상품 등록 API 구현

NaMinhyeok committed:
feat : 상품 ID로 상품 상세정보 조회 API 구현

NaMinhyeok committed:
feat : 상품 ID로 상품 삭제 API 구현

하지만 현재 상품을 삭제할 때 상품이 없으면 예외를 던지게 하기 위해 Select를 먼저 하고있음 이를 해결 할 수 있는 방법이 존재할까 ?

NaMinhyeok committed:
feat : 상품 ID로 상품 수정 구현

NaMinhyeok committed:
refactor : import문 정리 및 에러메시지 수정

NaMinhyeok committed:
fix: 테스트 깨지는 부분 해결

autoincrement id 에 대한 테스트가 문제였던 것 같습니다 꺼내서 사용하도록 테스트 방식을 바꾸었습니다
제대로된 방식인지는 모르겠지만 이 방법 이외에는 아이디어가 없어서 작성하였습니다

NaMinhyeok committed:
Merge pull request #4 from NaMinhyeok/feat/상품API-구현

[feat] 상품api 구현

NaMinhyeok committed:
feat: 주문 생성 API 구현


- 요청에 상품 id와 수량을 받로고 요청 수정

NaMinhyeok committed:
refactor: create 대신 builder를 사용하도록 변경

- 하위패키지에서 상위패키지 의존성을 지우기 위함

NaMinhyeok committed:
feat: 주문 조회 API 구현

NaMinhyeok committed:
feat: 이메일로 주문 목록 조회 API 구현

NaMinhyeok committed:
feat: 14시 이전 주문건 발송처리 구현

NaMinhyeok committed:
test: 주문 레포지토리 테스트

NaMinhyeok committed:
feat: 주문 상태 변경 비동기로 처리

비동기 처리시 테스트에서 오류가 발생한다
비동기처리를 동기처리로 변경하여서 테스트를 진행한다

NaMinhyeok committed:
test: 테스트 진행시 ID가져오는 부분 수정

개별 테스트는 ID를 지정해도 되나 전체 테스트가 진행 될 때 id는 get을 통해 사용해야한다

NaMinhyeok committed:
refactor: 사용하지 않는 import 정리

NaMinhyeok committed:
Merge pull request #5 from NaMinhyeok/feat/주문API-구현

[feat] 주문 API 구현

NaMinhyeok committed:
refactor: 주소, 우편번호 객체화

Embaddable 사용해서 객체화 진행

NaMinhyeok committed:
refactor: order_item 테이블 정규화

Product를 통해서 사용하면 된다고 판단하여 정규화 진행하였습니다

NaMinhyeok committed:
refactor: 변수명 및 접근제한자 수정

빌더의 접근제한자 수정 및 의미있는 변수명을 가지도록 수정

NaMinhyeok committed:
refactor: Exception 수정

NoSuchElementException 보다 IllegalArgumentException이 더 정확한 표현같아서 수정하였습니다

NaMinhyeok committed:
Merge pull request #6 from NaMinhyeok/refactor/도메인-객체-수정

[refactor] 도메인 객체 수정 및 자잘한 수정

AnTaeho commented on src/main/java/gc/cafe/api/ApiResponse.java:
```diff
@@ -0,0 +1,33 @@
+import org.springframework.http.HttpStatus;
+
+@Getter
+public class ApiResponse<T> {
+
+    private int code;
+    private HttpStatus status;
+    private String message;
+    private T data;
+
+    public ApiResponse(HttpStatus status, String message, T data) {
+        this.code = status.value();
+        this.status = status;
+        this.message = message;
+        this.data = data;
+    }
+
+    public static <T> ApiResponse<T> of(HttpStatus httpStatus, String message, T data) {
+        return new ApiResponse<>(httpStatus, message, data);
+    }
+
+    public static <T> ApiResponse<T> of(HttpStatus httpStatus, T data) {
+        return of(httpStatus, httpStatus.name(), data);
+    }
+
+    public static <T> ApiResponse<T> ok(T data) {
+        return of(HttpStatus.OK, data);
+    }
+
+}
```
응답 바디의 규격을 정해주는 클래스가 맞나요?

NaMinhyeok replied:
공통응답을 정의해주기 위해서 작성한 클래스입니다 !

NaMinhyeok replied:
> 응답 바디의 규격을 정해주는 클래스가 맞나요?

공통응답을 정의해주기 위해서 작성한 클래스입니다 !

AnTaeho commented on src/main/java/gc/cafe/api/controller/order/OrderController.java:
```diff
@@ -0,0 +1,35 @@
+
+    private final OrderService orderService;
+
+    @PostMapping("/new")
+    public ResponseEntity<ApiResponse<OrderResponse>> createOrder(@Valid @RequestBody OrderCreateRequest request) {
+        return ResponseEntity.ok(ApiResponse.ok(orderService.createOrder(request.toServiceRequest())));
+    }
```
RESTful 한 API 를 설계할 때 URI는 자원을 가르키고 행동은 HTTP method로 저정하기 때문에 
/new 를 붙히지 않아도 될 것 같습니다! 
POST라는 method 자체가 그런 의미를 가지고 있으니까요!

추가로 ResponseEntity와 ApiResponse가 거의 비슷한 역할을 하는 것 같고 
~~.ok(~~.ok()) 이렇게 ok가 2중으로 작성되어야 하는거 같은데,
혹시 이런 부분은 하나로 통일할 수 있는 부분은 없었나요?

NaMinhyeok replied:
> RESTful 한 API 를 설계할 때 URI는 자원을 가르키고 행동은 HTTP method로 저정하기 때문에 /new 를 붙히지 않아도 될 것 같습니다! POST라는 method 자체가 그런 의미를 가지고 있으니까요!
> 
> 추가로 ResponseEntity와 ApiResponse가 거의 비슷한 역할을 하는 것 같고 ~.ok(~.ok()) 이렇게 ok가 2중으로 작성되어야 하는거 같은데, 혹시 이런 부분은 하나로 통일할 수 있는 부분은 없었나요?

new는 안붙이는게 맞는거같아요 PR 날리고 나서 아차했습니다 ㅠㅠ

ResponseEntity랑 공통응답은 통일 하려고 합니다. 지금 당장은 원래는 ApiResponse만 생각하고있었는데 ResponseEntity에 담아야겠다는 생각이 들어서 일단 담기만 했습니다.. 그냥 body에 담는다거나 하는 방식으로 변경이 필요할거같아요 !

지금 당장은 이 부분보다 다른 부분에 집중하느라 놓쳤던 것 같습니다. 리팩토링 해보도록 하겠습니다 !

AnTaeho commented on src/main/java/gc/cafe/api/controller/order/request/OrderCreateRequest.java:
```diff
@@ -0,0 +1,50 @@
+@Getter
+public class OrderCreateRequest {
+
+    @Email(message = "이메일 형식이어야 합니다.")
+    @Size(max = 50, message = "이메일은 50자 이하여야 합니다.")
+    @NotBlank(message = "이메일은 필수입니다.")
+    private String email;
+
+    @Size(max = 200, message = "주소는 200자 이하여야 합니다.")
+    @NotBlank(message = "주소는 필수입니다.")
+    private String address;
+
+    @Size(max = 20, message = "우편번호는 20자 이하여야 합니다.")
+    @NotBlank(message = "우편번호는 필수입니다.")
+    private String postcode;
+
+    @Valid
+    @NotNull(message = "주문 할 상품은 필수입니다.")
+    private List<OrderProductQuantity> orderProductsQuantity;
```
간단하게 만들다보니 저는 Validation 까먹었네요...
굳!

AnTaeho commented on src/main/java/gc/cafe/api/controller/order/request/OrdersByEmailRequest.java:
```diff
@@ -0,0 +1,15 @@
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+public class OrdersByEmailRequest {
+
+    private String email;
+
+    @Builder
+    private OrdersByEmailRequest(String email) {
+        this.email = email;
+    }
+}
```
모든 DTO에 record를 사용하지 않고, 
@Getter랑 @Builder를 활용하셨는데 이유가 있을까요?

NaMinhyeok replied:
> 모든 DTO에 record를 사용하지 않고, @getter랑 @builder를 활용하셨는데 이유가 있을까요?

아이코 근데 이거는 사용되지 않는 클래스네요 생각해보니까 일단 이유는 까먹었습니다.. 사용해야한다는것을 다른것에 중점을 두다보니.. 레코드의 존재를 잊고있었습니다.

레코드는 제가 따로 공부할 때 사용해보려고 했었는데요 편하긴 한거같은데 아직 익숙치 않아서 그렇다고 할까요 클래스 옆에 파라미터로 적는게 익숙하지가 않아서 자꾸 습관적으로 클래스를 만들게되더라구요

태호님 말대로 record를 사용하면 builder를 사용할 필요도 없고 getter도 안써도 되니 record로 전부 리팩토링 진행해도 좋을 것 같네요! 감사합니다

AnTaeho commented on src/main/java/gc/cafe/api/controller/product/request/ProductCreateRequest.java:
```diff
@@ -0,0 +1,47 @@
+        this.description = description;
+    }
+
+    public ProductCreateServiceRequest toServiceRequest() {
+        return ProductCreateServiceRequest.builder()
+            .name(name)
+            .category(category)
+            .price(price)
+            .description(description)
+            .build();
+    }
```
클라이언트 - 컨트롤러 / 컨트롤러 - 서비스
이 두 계층간의 이동때 각각의 DTO를 지정해주셨는데 
저도 이 부분은 잘 모르는데 혹시 특별한 이유가 있나요? 

예전에 어디서 이렇게 해야된다고 들은거 같기도 해서요

NaMinhyeok replied:
> 클라이언트 - 컨트롤러 / 컨트롤러 - 서비스 이 두 계층간의 이동때 각각의 DTO를 지정해주셨는데 저도 이 부분은 잘 모르는데 혹시 특별한 이유가 있나요?
> 
> 예전에 어디서 이렇게 해야된다고 들은거 같기도 해서요

이거 저도 그냥 어디서 듣고 따라한건데요 패키지간의 의존성이 생기면 안된다고 합니다.

https://www.youtube.com/watch?v=dJ5C4qRqAgA&t=1028s

이 영상이 그에 대한 완전한 답은 아닐수도 있지만 여기서도 패키지간 의존성이라는 이야기가 나옵니다.
그리고 MVC 패턴에서 Model은 view와 controller에서 사용되어도 되지만 view에서는 controller가 사용되면 안되는 것 처럼
비즈니스 레이어인 Service와 Persistance 레이어 맞나요 ? 하여튼 컨트롤러에서 클래스의 분리가 되어야 된다고 생각했습니다.

한마디로 말하자면 서비스는 컨트롤러를 몰라야한다는거죠 하지만 서비스에서 ProductCreateRequest를 그대로 사용하면 서비스는 컨트롤러 레이에어 의존하고 있다 라고 볼수도 있을 것 같아요 그래서 컨트롤러에서 서비스로 데이터를 내려주도록 하였습니다.

AnTaeho commented on src/main/java/gc/cafe/api/service/order/response/OrderResponse.java:
```diff
@@ -0,0 +1,46 @@
+        this.orderDetails = orderDetails;
+    }
+
+    public static OrderResponse of(Order order) {
+        return OrderResponse.builder()
+            .id(order.getId())
+            .email(order.getEmail())
+            .address(order.getAddress().getAddress())
+            .postcode(order.getAddress().getPostcode())
+            .orderStatus(order.getOrderStatus())
+            .orderDetails(order.getOrderProducts().stream()
+                .map(orderProduct -> OrderDetailResponse.builder()
+                    .category(orderProduct.getProduct().getCategory())
+                    .price(orderProduct.getProduct().getPrice())
+                    .quantity(orderProduct.getQuantity())
+                    .build())
+                .toList())
+            .build();
+    }
```
.map() 들어가는 부분은 메서드 추출해주면 가독성이 좀 더 좋아질 것 같습니다!

AnTaeho commented on src/main/java/gc/cafe/api/service/product/ProductServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+        return ProductResponse.of(saveProduct);
+    }
+
+    @Override
+    public Long deleteProduct(Long id) {
+        productRepository.findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("해당 id : " + id + "를 가진 상품을 찾을 수 없습니다."));
+        productRepository.deleteById(id);
+        return id;
+    }
```
정확한 기억은 아닌데 deleteById를 하면 한번 findById (existsById 일수도 있고요 )를 해서 존재하는지 찾고 삭제해주는걸로 알고 있습니다!

https://hwanchang.tistory.com/7

NaMinhyeok replied:
> 정확한 기억은 아닌데 deleteById를 하면 한번 findById (existsById 일수도 있고요 )를 해서 존재하는지 찾고 삭제해주는걸로 알고 있습니다!
> 
> https://hwanchang.tistory.com/7

앗 그렇네요 그쪽에서 예외처리를 진행하도록 하면 될거같네요 예전엔 분명 이렇게 안했던거같은데 갑자기 예외처리를 어디서 해야되지 라는 생각에.. 섣부르게 작성한것같습니다 감사합니다

AnTaeho commented on src/main/java/gc/cafe/domain/BaseEntity.java:
```diff
@@ -0,0 +1,22 @@
+@EntityListeners(AuditingEntityListener.class)
+public abstract class BaseEntity {
+
+    @CreatedDate
+    private LocalDateTime createdAt;
```
```suggestion
    @Column(updatable = false)
    @CreatedDate
    private LocalDateTime createAt;
```

생성 일자는 업데이트 못하게 막아둘 수 있습니다!

AnTaeho commented on src/main/java/gc/cafe/domain/order/Order.java:
```diff
@@ -0,0 +1,56 @@
+    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
+    private List<OrderProduct> orderProducts = new ArrayList<>();
+
+    @Builder
+    private Order(String email, String address, String postcode, List<Product> products, Map<Long,Integer> orderProducts) {
+        this.email = email;
+        this.address = Address.builder()
+            .address(address)
+            .postcode(postcode)
+            .build();
+        this.orderStatus = OrderStatus.ORDERED;
+        this.orderProducts = products.stream()
+            .map(product -> new OrderProduct(this, product, orderProducts.get(product.getId())))
+            .collect(Collectors.toList());
+    }
```
빈 리스트로 초기화 되어 있으니까 한번에 뭉쳐서 설정해주기보단 
OrderProduct가 생성되는 시점에 생성자에서 처리해주면 어떨까 생각합니다.

https://skatpdnjs.tistory.com/49

NaMinhyeok replied:
> 빈 리스트로 초기화 되어 있으니까 한번에 뭉쳐서 설정해주기보단 OrderProduct가 생성되는 시점에 생성자에서 처리해주면 어떨까 생각합니다.
> 
> https://skatpdnjs.tistory.com/49

조금 이해가 안되어서 상세하게 풀어서 설명해주실 수 있을까요 ?

제가 이해한 바로는 Order를 생성 할 때 OrderProduct까지 생성해버리지 말고 OrderProduct를 저장하는 메서드를 처리하라는걸까요 ?

AnTaeho commented on src/main/java/gc/cafe/domain/order/OrderRepository.java:
```diff
@@ -0,0 +1,14 @@
+
+import java.util.List;
+
+@Repository
+public interface OrderRepository extends JpaRepository<Order, Long> {
+
+    List<Order> findByEmail(String email);
+
+    List<Order> findByOrderStatus(OrderStatus orderStatus);
+}
```
제가 알기로 JPARepository 타고 타고 어딘가에 @Repository 가 있어서 반드시 붙혀줄 필요는 없는 거로 압니다! 
리포지토리임을 명시할 목적이라면 있어도 상관 없을 것 같고요

NaMinhyeok replied:
> 제가 알기로 JPARepository 타고 타고 어딘가에 @repository 가 있어서 반드시 붙혀줄 필요는 없는 거로 압니다! 리포지토리임을 명시할 목적이라면 있어도 상관 없을 것 같고요

저도 명시하지 않아도 된다는 사실은 알고있었는데 이건 그냥 제가 불편해서 붙였습니다..
엔티티(`@entity`) - 레포지토리 - 서비스(`@service`) - 컨트롤러(`@RestController`)
다 하나씩 붙이는데 얘만 없는게 허전하다고할까요... 물론 실제로 필요한 어노테이션과 필요없는 어노테이션은 있지만 조회 메서드에 `@Transactional(readonly = true)` 붙이는 것과 같이 의도적으로 명시하고 싶었습니다

이건 순전히 제 취향인거같습니다 ! 감사합니다 !

AnTaeho reviewed (commented):
고생하셨습니다. 
몇가지 의견 남겼습니다! 
그리고 리뷰는 안달았는데 테스트 코드 진짜 빡세게 작성하셨네여...

jmd5314 commented on src/main/java/gc/cafe/api/ApiControllerAdvice.java:
```diff
@@ -0,0 +1,22 @@
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+
+@RestControllerAdvice
+public class ApiControllerAdvice {
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(BindException.class)
+    public ApiResponse<Object> bindException(BindException e) {
+        return ApiResponse.of(
+            HttpStatus.BAD_REQUEST,
+            e.getBindingResult().getAllErrors().get(0).getDefaultMessage(),
+            null
+        );
+    }
+}
```
저는 유효 검증 처리를 안해놔서 할 말이 없긴 한데, @Valid에서 발생하는 예외 (MethodArgumentNotValidException)에 대한 처리를 여기에 추가해주면 좀 더 좋지 않을까 합니다

NaMinhyeok replied:
> 저는 유효 검증 처리를 안해놔서 할 말이 없긴 한데, @Valid에서 발생하는 예외 (MethodArgumentNotValidException)에 대한 처리를 여기에 추가해주면 좀 더 좋지 않을까 합니다

감사합니다 ! 참고해서 리팩토링 해보도록 하겠습니다 공통응답부분은 제가 많이 부족하기도 하고 단순히 가벼운 포맷을 만든 것이라서 아쉬운 부분이 많았네요 참조해서 도전해보겠습니다 !

NaMinhyeok replied:
공부하면서 찾아본 내용으로는 BindException이 더 넓은 범위를 커버한다고 하네요 ! MethodArgumentNotValidException을 타고 들어가서 확인하니 extend로 BindException을 상속받고 있는 모습을 볼 수 있습니다. 그래서 더 상세하게 에러를 나눌 부분이 존재 할 때 나누면 좋을 것 같습니다 ! 

저도 Valid에 대해서 그냥 이런게 있구나 하고 가볍게 사용만 해봤었는데 덕분에 공부하게 되었습니다. 이미 알고 계실 수도 있지만 아래의 글 참고해보시면 좋을 것 같습니다 !

https://velog.io/@park2348190/Spring%EC%9D%98-Validation

jmd5314 replied:
오.. BindException이 더 상위 개념인줄은 몰랐네요 민혁님 덕분에 좋은 정보 알아갑니다 
저는 `@Email` 이나 `@NotNull` 같은 기본적인 어노테이션만 대충 써봤었는데, 생각보다 
제약 조건에 관한 애노테이션이 되게 많네요..
커스텀 애노테이션 만들어서 적용하는 부분도 있던데 수정하면서 한번 구현해 봐야겠습니다 
글 링크 감사합니다!

jmd5314 commented on src/main/java/gc/cafe/api/ApiResponse.java:
```diff
@@ -0,0 +1,33 @@
+        this.data = data;
+    }
+
+    public static <T> ApiResponse<T> of(HttpStatus httpStatus, String message, T data) {
+        return new ApiResponse<>(httpStatus, message, data);
+    }
+
+    public static <T> ApiResponse<T> of(HttpStatus httpStatus, T data) {
+        return of(httpStatus, httpStatus.name(), data);
+    }
```
이건 제가 잘 몰라서 궁금한 부분인데 message가 없을 경우를 고려해서 두개로 작성하신건가요??

NaMinhyeok replied:
> 이건 제가 잘 몰라서 궁금한 부분인데 message가 없을 경우를 고려해서 두개로 작성하신건가요??

제가 제대로 구현하지 못한 부분같은데요 그냥 못짠거같아요 ㅋㅋㅋㅋ 아래 ok 리턴하기 위해서 사용하고있는데 이거는 진짜 좀 더 공통응답에 대해서 생각해볼 필요가있는거같아요 아무래도 공통응답쪽은 대체적으로 코드가 별로이긴합니다. 다 ok가거나 400이 가도록 되어있어서 리팩토링 해보도록하겠습니다 !

jmd5314 commented on src/main/java/gc/cafe/api/service/order/OrderServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+    private final ProductRepository productRepository;
+
+    @Override
+    public OrderResponse createOrder(OrderCreateServiceRequest request) {
```
이렇게 서비스 와 컨트롤러 사이의 dto를 만들어서 구현하면 컨트롤러와 서비스 간의 의존성을 줄이지만 엄청 번거롭다는 글을 봤었는데, 이걸 진짜 적용하셨네요..
이번 과제 진짜 빡세고 꼼꼼하게 구현하신거 같습니다..!

NaMinhyeok replied:
> 이렇게 서비스 와 컨트롤러 사이의 dto를 만들어서 구현하면 컨트롤러와 서비스 간의 의존성을 줄이지만 엄청 번거롭다는 글을 봤었는데, 이걸 진짜 적용하셨네요.. 이번 과제 진짜 빡세고 꼼꼼하게 구현하신거 같습니다..!

감사합니다 ! 저는 막상 이 얘길 들었을 때 아차했거든요 한번 적용해보셔도 좋을 것 같아요 !

jmd5314 commented on src/test/java/gc/cafe/docs/order/OrderControllerDocsTest.java:
```diff
@@ -0,0 +1,318 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class OrderControllerDocsTest extends RestDocsSupport {
```
진짜 고생하셨습니다..!

NaMinhyeok replied:
> 진짜 고생하셨습니다..!

믿음님은 스웨거를 이용하셨던데 현업에서 REST Docs를 많이 사용하는 추세라고 하니 직접 작성은 아니더라도 이런내용이 있구나 하고 봐주셔도 좋을 것 같습니다 !

jmd5314 reviewed (commented):
코드 잘봤습니다
진짜 빡세게 문서화 하고 테스트 구현 하신거 같네요
제가 크게 도움은 못 될거 같고 사소한 리뷰 정도만 달아봤습니다
고생하셨습니다!

NaMinhyeok committed:
refactor: RESTful 한 API 설계를 위해 new 삭제

NaMinhyeok committed:
refactor: 공통응답 수정

- ResponseEntity를 사용하지 않도록 수정하였습니다
- created를 추가하여 상태코드 201을 받을 수 있도록 하였습니다.
- IllegalArgumentException도 처리 할 수 있도록 하였습니다

NaMinhyeok committed:
refactor: deleteById를 findById, delete 로 수정

deleteById는 어차피 select 쿼리를 내부적으로 보내고있으니 명시적으로 select 쿼리를 보내고 없으면 예외를 발생시키고 삭제하도록 구현

NaMinhyeok committed:
refactor: 코드에 영향 없는 코드컨벤션, 사용하지않는 import 제거, 메서드 추출 등 리팩토링

NaMinhyeok committed:
refactor: 연관관계 편의 메서드를 이용하여 구현

- cascade all -> remove 로 변경 기존 코드는 cascade에 persist가 있어야지만 order_items 테이블에 생성되었음
- 생성 방식에 따라 test 코드 수정
- 서비스레이어 테스트에서 생성시 order_items도 들어가는지 확인

NaMinhyeok committed:
refactor: 변경 불가능 컬럼 지정, 접근 지정자 수정

NaMinhyeok committed:
test: 연관관계 편의메서드 사용으로 인한 테스트 수정

NaMinhyeok committed:
refactor: 코드에 영향없는 import 정리

NaMinhyeok committed:
Merge pull request #7 from NaMinhyeok/refactor/리뷰내용-리팩토링

[refactor] 코드리뷰 내용 반영
